### DISCUSSION TITLE
Never sure what method to use for caught exceptions or informational type messages?

### DISCUSSION THREAD

**Discussion by ColtonIdle** [getsentry/sentry-java#3465] [General]
Still sorta new to sentry.

I see

Sentry.captureException
Sentry.captureMessage
Sentry.captureEvent

I know don't understand when to use each, when. can anyone help disambiguate? thank you!

**Comment by romtsn** [comment]
> Sentry.captureException

For custom or handled exceptions, works the same way as our automatic uncaught exception handler pretty much, but you capture it manually in this case. 

> Sentry.captureMessage

For plain string informational messages or logs, e.g. if you want to get alerted on some kinds of logs, etc.

> Sentry.captureEvent

For full-fledged event with everything under your control. All the values you set will be considered and not overridden by the global values (e.g. tags, breadcrumbs, contexts, etc.). Requires more boilerplate from your side.

Essentially `captureMessage` and `captureException` are just shortcuts for `captureEvent` provided to your convenience. Does that make sense?

**Comment by ColtonIdle** [comment]
Amazing. Thank you so much

Sent from Proton Mail Android

-------- Original Message --------
